docs/markdown/Docker/docker.md

```diff
@@ -61,31 +61,39 @@ docker_image(
 
-Pants, however, takes care of assembling the context for you. It does so using the dependencies of the `docker_image` target. 
+Pants, however, takes care of assembling the context for you. It does so using the dependencies of the [`docker_image`](doc:reference-docker_image) target, which can include:
 
-A [`docker_image`](doc:reference-docker_image) can depend on loose files belonging to [`file` / `files` targets](doc:resources#files), and on artifacts packaged from a variety of targets, such as [`pex_binary`](doc:reference-pex_binary) , [`python_distribution`](doc:reference-python_distribution), [`archive`](doc:reference-archive), or any other target that can be built via the [package](doc:reference-package) goal.
+- Loose files specified using  [`file` / `files` targets](doc:resources#files).
+- Artifacts packaged from a variety of targets, such as [`pex_binary`](doc:reference-pex_binary) , [`python_distribution`](doc:reference-python_distribution), [`archive`](doc:reference-archive), and any other target that can be built via the [package](doc:reference-package) goal, including other docker images.
 
 The context is assembled as follows:
 
 - The sources of `file` / `files` targets are assembled at their relative path from the repo root.
-- The artifacts of any packageable targets are built, as if by running `./pants package`, and placed in the context using the artifact's `output_path` field.
+- The artifacts of any packaged targets are built, as if by running `./pants package`, and placed in the context using the artifact's `output_path` field.
   - The `output_path` defaults to the scheme `path.to.directory/tgt_name.ext`, e.g. `src.python.helloworld/bin.pex`.
 
-### Dependency inference for `pex_binary`
+### Dependency inference support
 
 When you `COPY` PEX binaries into your image, the dependency on the `pex_binary` target will be inferred, so you don't have to add that explicitly to the list of `dependencies` on your `docker_image` target.
 
-For example, the `pex_binary` target `src/python/helloworld/bin.pex` has the default `output_path` of `src.python.helloworld/bin.pex`. So, Pants can infer a dependecy based on the line `COPY src.python.helloworld/bin.pex /bin/helloworld`.
+For example, the `pex_binary` target `src/python/helloworld/bin.pex` has the default `output_path` of `src.python.helloworld/bin.pex`. So, Pants can infer a dependency based on the line `COPY src.python.helloworld/bin.pex /bin/helloworld`.
+
+Inference for Go binaries and artifacts of other packaged targets is similar.
+
+Inference is also supported for `docker_image` targets specified in build arguments, for example:
+
+```dockerfile
+ARG BASE_IMAGE=:base
```

Comment:
I think it would be worth making it very clear that `:base` here is a target address, and that Pants will provide the built docker image name for that address as the `BASE_IMAGE` build arg to the `docker build` command.

Reply:
Was hoping you would clarify :-)  Updated now.